### DISCUSSION TITLE
Unable to connect to MySQL server - Django & Github Actions

### DISCUSSION THREAD

**Discussion by RidenShark** [community/community#25967] [Actions]
<p>I’m trying to setup github actions for my django project.  I am unable to connect to the service. <code>django.db.utils.OperationalError: (2003, "Can't connect to MySQL server on '127.0.0.1' (111)")</code>.</p>
<p>This is my workflow config.</p>
<pre><code class="lang-auto">name: Myapp CI

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest
    
    services:
      mysql:
        image: mysql:5.7
        env:
            MYSQL_ALLOW_EMPTY_PASSWORD: yes
            MYSQL_DATABASE: myapp
        ports:
            - 3306
        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
    
    strategy:
      max-parallel: 4
      matrix:
        python-version: [3.8]

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v1
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install Dependencies
      run: |
        python -m pip install --upgrade pip
        pip install -r requirements-dev.txt
    - name: Run Tests
      run: |
        python manage.py test
</code></pre>
<p>It’s been a whole day and I still couldn’t figure out what’s wrong with the MySQL setup. Can anyone help? Is there an official documentation on how to setup MySQL on github actions?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>If you give only one port number the port number on the host (what you can reach at localhost/127.0.0.1/::1) is chosen randomly, <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-localhost" rel="nofollow noopener">as shown in the services example</a>. Could that be the issue?</p>
<p>If you want the MySQL service to be available on 127.0.0.1:3306 your <code>ports</code> setting should look like this:</p>
<pre><code class="lang-yaml">ports:
  - '3306:3306'
</code></pre>

**Comment by RidenShark** [comment]
<p>:pleading_face: Thank you! I am new and learning and didn’t know about this. Spent almost a day trying!</p>

**Comment by csaba-kovacs** [comment]
I have the same issue but the port is set correctly.

**Reply by airtower-luna** [reply]
Are you possibly using a job container? In that case the port mapping won't be available (because it's on the host VM), and you have to connect to the service container by name.
